src/core/server/http/http_server.test.ts

```diff
@@ -288,6 +295,94 @@ test('valid body', async () => {
     });
 });
 
+test('valid body with validate function', async () => {
+  const router = new Router('/foo', logger, enhanceWithContext);
+
+  router.post(
+    {
+      path: '/',
+      validate: {
+        body: new RouteValidator(({ bar, baz } = {}) => {
```

Comment:
I'm a bit concerned with the amount of code that users have to write. If you want to declare a request handler separately it doesn't become simpler:
```ts
const handler: RequestHandler<never, never, RouteValidateSpecs<Stream>> = (context, req, res) => {
   req.body
```
What do you think if we always require users to specify types explicitly? Something like:
```ts
const validate = {
  query: schema.object({
    page: schema.number(),
  }),
  body: schema.stream(),
};
router.post( {
  path: '/',
  validate,
  // or another type helper that we provide
  <RequestTypeOf validate>(context, req, res) =>
});
```

Reply:
Thanks for pointing that out!
I've revisited the type inference and I think I can get to the more sensible point:

```ts
const handler: RequestHandler<unknown, unknown, Stream> = (context, req, res) => {
  req.body // This is a Stream
}
```

`RequestHandler` is not (should not) be tied to the Config-Schema/RouteValidation types but to their outputs.

TS compiles. Running tests locally and, if all good, I'll push the changes :)